src/include/duckdb/execution/index/art/node.hpp

```diff
@@ -10,18 +10,34 @@
 
 #include "duckdb/execution/index/art/art_key.hpp"
 #include "duckdb/common/common.hpp"
+#include "duckdb/storage/meta_block_writer.hpp"
+#include "duckdb/storage/meta_block_reader.hpp"
+#include "duckdb/storage/index.hpp"
 
 namespace duckdb {
 enum class NodeType : uint8_t { N4 = 0, N16 = 1, N48 = 2, N256 = 3, NLeaf = 4 };
-
 class ART;
 
+struct SwizzleablePointer {
+	~SwizzleablePointer() {};
+	SwizzleablePointer(idx_t block_id, idx_t offset);
+	SwizzleablePointer() : pointer(0) {};
+	uint64_t pointer;
+	SwizzleablePointer &operator=(const uint64_t &ptr);
+	friend bool operator!=(const SwizzleablePointer &s_ptr, const uint64_t &ptr);
+
+public:
+	DiskPosition GetSwizzledBlockInfo();
+	bool IsSwizzled();
+};
+
 class Node {
 public:
 	static const uint8_t EMPTY_MARKER = 48;
+	duckdb::Deserializer *source;
```

Comment:
Also it would be great if we could avoid mixing static variables and member variables in the same block, as it can become quite unclear what is a member variable and what is not.

Reply:
what you mean with same block?

Reply:
e.g.:

```
class Node {
public:
   Node();

private:
   duckdb::Deserializer *source;

 private:
   void MyFunction();

};